code/game/objects/items/weapons/cigs.dm

```diff
@@ -100,58 +107,159 @@ LIGHTERS ARE IN LIGHTERS.DM
 	if(!lit)
 		light("<span class='warning'>[src] is lit by the flames!</span>")
 
-/obj/item/clothing/mask/cigarette/welder_act(mob/user, obj/item/I)
-	. = TRUE
-	if(I.tool_use_check(user, 0)) //Don't need to flash eyes because you are a badass
-		light("<span class='notice'>[user] casually lights [src] with [I], what a badass.</span>")
+/obj/item/clothing/mask/cigarette/proc/lighter_interaction(obj/item/I, mob/user, mob/living/carbon/target)
+	if(lit)
+		to_chat(user, "<span class='warning'>\The [initial(name)] is already lit!</span>")
+		return FALSE
 
-/obj/item/clothing/mask/cigarette/attackby(obj/item/I, mob/user, params)
-	..()
 	if(istype(I, /obj/item/lighter/zippo))
-		var/obj/item/lighter/zippo/Z = I
-		if(Z.lit)
-			light("<span class='rose'>With a single flick of [user.p_their()] wrist, [user] smoothly lights [user.p_their()] [name] with [user.p_their()] [Z]. Damn [user.p_theyre()] cool.</span>")
-
-	else if(istype(I, /obj/item/lighter))
-		var/obj/item/lighter/L = I
-		if(L.lit)
-			light("<span class='notice'>After some fiddling, [user] manages to light [user.p_their()] [name] with [L].</span>")
-
-	else if(istype(I, /obj/item/match/unathi))
-		var/obj/item/match/unathi/U = I
-		if(U.lit)
-			playsound(user.loc, 'sound/effects/unathiignite.ogg', 40, FALSE)
-			light("<span class='rose'>[user] spits fire at [user.p_their()] [name], igniting it.</span>")
-			U.matchburnout()
+		var/obj/item/lighter/zippo/Zip = I
+		if(Zip.lit)
+			if(target == user)
+				light("<span class='rose'>With a single flick of [user.p_their()] wrist, [user] smoothly lights [user.p_their()] [name] with [I]. Damn [user.p_theyre()] cool.</span>")
+			else
+				light("<span class='rose'>[user] whips [I] out and holds it for [target]. [user.p_their(TRUE)] arm is as steady as the unflickering flame [user.p_they()] lights \the [name] with. Damn [user.p_theyre()] cool.</span>")
+			return TRUE
+		return failed_to_light(I, user)
+
+	if(istype(I, /obj/item/match))
+		var/obj/item/match/Match = I
+		if(Match.lit)
+			if(target == user)
+				light("<span class='notice'>[user] lights [user.p_their()] [name] with [I].</span>")
+			else
+				light("<span class='notice'>[user] holds [I] out for [target], and lights \the [name].</span>")
```

Comment:
```suggestion
				light("<span class='notice'>[user] holds [I] out for [target], and lights [name].</span>")
```

Reply:
the is actually required or it will say "lights cigarette"